content/concepts/discovery/mDNS.md

```diff
@@ -0,0 +1,47 @@
+---
+title: "mDNS"
+description: "MDNS uses a multicast system of DNS records over a local network to enable peer discovery."
+weight: 224
+---
+
+## What is mDNS?
+
+mDNS, or multicast Domain Name System, is a way for nodes to use a multicast system
+of DNS records over a local network to discover and communicate with nodes. Nodes
+broadcast topics they're interested in instead of querying a central name server.
+The discovery, however, is limited to the peers in the local network. mDNS is commonly
+used on home networks to allow devices such as computers, printers, and smart TVs to
+find each other and connect. It uses a protocol called multicast to broadcast messages
+on the network, allowing devices to discover each other and exchange information.
+
+## mDNS in libp2p
+
+In libp2p, mDNS is used for service discovery.
+When a peer wants to connect to another peer in a libp2p network,
+it can use mDNS to resolve the hostname of the other peer to its multiaddr.
+A peer can broadcast a query request to *find all peers* on a local network to
+receive DNS response messages from peers, which contain the peer information
+of discovered peers. The response message is in the form of a DNS record:
+
+`<service-name> PTR <peer-name>.<service-name>`,
+
+where `<service-name>` is `_p2p._udp.local`, the name of the service that is being
+advertised.
+> `peer-name` is not used for anything and can be filled with a string with random characters.
```

Comment:
This needs to be rephrased. Example:
In libp2p, we don't have any use for `peer-name`. Since the wire format requires us to send something, we send a string of random ASCII characters.